cfg/agentinfo/info_test.go

```diff
@@ -80,42 +81,79 @@ func TestPlugins(t *testing.T) {
 	OutputPlugins = []string{"x", "y", "z"}
 
 	isRunningAsRoot = func() bool { return true }
-	plugins := Plugins()
+	plugins := Plugins("")
 	expected := fmt.Sprintf("inputs:(a b c) outputs:(x y z)")
 	if plugins != expected {
 		t.Errorf("wrong plugins string constructed '%v', expecting '%v'", plugins, expected)
 	}
 
 	isRunningAsRoot = func() bool { return false }
-	plugins = Plugins()
+	plugins = Plugins("")
 	expected = fmt.Sprintf("inputs:(a b c run_as_user) outputs:(x y z)")
 	if plugins != expected {
 		t.Errorf("wrong plugins string constructed '%v', expecting '%v'", plugins, expected)
 	}
+
+	plugins = Plugins("/aws/ecs/containerinsights/ecs-cluster-name/performance")
+	expected = fmt.Sprintf("inputs:(a b c run_as_user) outputs:(x y z container_insights)")
+	if plugins != expected {
+		t.Errorf("wrong plugins string constructed '%v', expecting '%v'", plugins, expected)
+	}
 }
 
 func TestUserAgent(t *testing.T) {
-	userAgent = ""
 	VersionStr = "VSTR"
 	BuildStr = "BSTR"
 	InputPlugins = []string{"a", "b", "c"}
 	OutputPlugins = []string{"x", "y", "z"}
 
 	isRunningAsRoot = func() bool { return true }
 
-	ua := UserAgent()
-	expected := fmt.Sprintf("CWAgent/VSTR (%v; %v; %v) BSTR inputs:(a b c) outputs:(x y z)", runtime.Version(), runtime.GOOS, runtime.GOARCH)
-	if ua != expected {
-		t.Errorf("wrong UserAgent string constructed '%v', expecting '%v'", ua, expected)
+	tests := []struct {
+		name         string
+		logGroupName string
+		expected     string
+		errorMessage string
+	}{
+		{
+			"non container insights",
+			"test-group",
+			fmt.Sprintf("CWAgent/VSTR (%v; %v; %v) BSTR inputs:(a b c) outputs:(x y z)", runtime.Version(), runtime.GOOS, runtime.GOARCH),
```

Comment:
Not sure if those print statement is necessary since this is just tests.

Reply:
It's not a print statement that goes to the console. It's a format string for the `expected` attribute in the test struct. I'm fine with it as is, since the value for the UserAgent output is derivable.